File: bubble.py
import time
import psutil
import platform
import cpuinfo
import subprocess


def bubble_sort(numeros):
    for i in range(len(numeros) - 1):
        for j in range(len(numeros) - 1 - i):
            if numeros[j] > numeros[j + 1]:
                numeros[j], numeros[j + 1] = numeros[j + 1], numeros[j]

def ler_arquivo(caminho):
    with open(caminho, 'r', encoding='utf-8-sig') as arquivo:
        return [int(linha.strip()) for linha in arquivo]

def arquivo_ordenado(caminho, numeros):
    with open(caminho, 'w') as arquivo:
        arquivo.writelines(f"{num}\n" for num in numeros)

def obter_info_lscpu():
        cpu_info = subprocess.check_output("lscpu", shell=True, text=True)
        return cpu_info

print(f"Linguagem: Python {platform.python_version()}")

info_sistema = platform.uname()
print(f"Sistema: {info_sistema.system} {info_sistema.release}")

print("processador:")

print(obter_info_lscpu())

print(f"Memória RAM total: {psutil.virtual_memory().total / (1024 ** 3):.2f} GB")  

entrada = 'arq-desafio.txt'
saida = 'ordenado_bubble_py.txt'

inicio_tempo = time.time()
processo = psutil.Process()

mem_inicio = processo.memory_info().rss / 1024
numeros = ler_arquivo(entrada)
bubble_sort(numeros)
arquivo_ordenado(saida, numeros)
mem_fim = processo.memory_info().rss / 1024

fim_tempo = time.time()

print(f"Números ordenados foram salvos em '{saida}'.")
print(f"Tempo de execução: {round((fim_tempo - inicio_tempo) * 1000, 2)} ms")
print(f"Memória utilizada: {round(mem_fim - mem_inicio, 2)} KB")

File: bubble.php
<?php
function bubbleSort(&$numeros) {
    $n = count($numeros);
    for ($i = 0; $i < $n - 1; $i++) {
        for ($j = 0; $j < $n - 1 - $i; $j++) {
            if ($numeros[$j] > $numeros[$j + 1]) {
                $temp = $numeros[$j];
                $numeros[$j] = $numeros[$j + 1];
                $numeros[$j + 1] = $temp;
            }
        }
    }
}

function lerArquivo($caminho) {
    $conteudo = file($caminho, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    return array_map('intval', $conteudo);
}

function arquivo_ordenado($caminho, $numeros) {
    file_put_contents($caminho, implode(PHP_EOL, $numeros));
}

function obterInfoProcessador() {
    return shell_exec('lscpu');
}

echo "Linguagem: PHP " . PHP_VERSION . "\n";
echo "Sistema: " . PHP_OS . " - " . php_uname('r') . "\n";

echo "Processador:\n" . obterInfoProcessador() . "\n";

$memoriaTotalKB = trim(shell_exec("grep MemTotal /proc/meminfo | awk '{print $2}'"));
$memoriaTotalGB = round($memoriaTotalKB / 1024 / 1024, 2);
echo "Memória RAM total: " . $memoriaTotalGB . " GB\n";

$arquivoEntrada = 'arq-desafio.txt';
$arquivoSaida = 'ordenado_bubble_php.txt';

$inicioTempo = microtime(true);
$memoriaInicial = memory_get_usage() / 1024;

$numeros = lerArquivo($arquivoEntrada);
bubbleSort($numeros);
arquivo_ordenado($arquivoSaida, $numeros);

$fimTempo = microtime(true);
$memoriaFinal = memory_get_usage() / 1024;

echo "Números ordenados salvos em '$arquivoSaida'.\n";
echo "Tempo de execução: " . round(($fimTempo - $inicioTempo) * 1000, 2) . " ms\n";
echo "Memória utilizada: " . round($memoriaFinal - $memoriaInicial, 2) . " KB\n";
?>
